### DISCUSSION TITLE
[Product] 사용자 정의 소비 카테고리 삭제 시나리오

### DISCUSSION THREAD

**Discussion by psychology50** [CollaBu/pennyway-was#91] [Q&A] [answered]
![image](https://github.com/CollaBu/pennyway-was/assets/96044622/f38b3609-6440-474f-a5de-f550a5415f54)

Pennyway 서비스에는 위와 같이 사용자가 원하는 소비 카테고리를 등록할 수 있는 기능이 제공됩니다.  

이러한 기능의 문제는 삭제에서 비롯하는데, 문제의 개요는 이렇습니다.
1. 사용자가 커스텀 카테고리를 생성한다.
2. 사용자가 새로운 소비 내역을 커스텀 카테고리에 등록한다.
3. 사용자가 커스텀 카테고리를 삭제한다.

<br/>

이렇게 진행이 되면 `(3)`의 결과로 인하여, `(2)`에서 등록한 새로운 소비 내역도 함께 삭제된다는 문제가 있습니다.  

따라서 위 문제에 대해서 어떻게 대응할 지 고민이 필요합니다.

<br/>

### 🤔 떠오르는 방안
1. 사용자가 커스텀 카테고리를 삭제하면, 해당 카테고리에 등록한 지출 내역도 함께 삭제한다.
     - 미리 사용자에게 삭제될 수 있음을 경고하는 알림을 띄울 수 있다.
2. 사용자가 커스텀 카테고리를 삭제하려고 하면, 해당 카테고리에 등록된 소비 내역의 카테고리를 모두 공백(null)으로 수정한다.
     - 이렇게 되면 사용자는 지출 항목의 소비처, 메모를 입력하지 않았을 시 소비 기록의 출처를 짐작할 수 없게 됨
     - 또한 서버에서 해당 데이터를 유용하게 사용할 수 없음.
2. 사용자가 커스텀 카테고리를 삭제하려고 하면, 다른 카테고리로 옮길 수 있도록 하거나 모두 삭제시킬 수 있는 선택권을 제공함.
     - UI와 시스템 플로우가 추가됨. (작업 딜레이)

위 방안 중 어떤 정책을 선택하는 것이 더 좋을 지, 혹은 더 좋은 아이디어가 있는 지 의견을 구하고자 합니다.

**Comment by heejinnn** [comment, marked as answer]
저는 1번 카테고리를 삭제하는 경우 해당 카테고리에 등록된 지출 내역도 함께 삭제되는 것이 좋다고 생각합니다. 
물론, 3번 옵션이 유저에게 선택권을 제공하여 좋지만, 작업 딜레이가 예상되기 때문에 배포 후 유저 경험을 통해 피드백을 받아보고 추후에 3번을 고려해도 된다고 생각합니다.

카카오맵은 카테고리가 아닌 그룹으로 관리하는 방식이지만 이 방식처럼, 카테고리를 삭제하면 카테고리 안의 모든 데이터가 삭제된다는 경고창을 보여주고

<img src = "https://github.com/CollaBu/pennyway-was/assets/103185302/b3c9a1b2-5602-425c-ac15-82234cbe4870" width = 300/>

<br/>

<br/>

경고창에 몇 건의 내역이 존재하는 지 여부를 함께 표시하여 삭제 건수를 확인할 수 있도록 하면 좋을 것 같습니다.

<img src = "https://github.com/CollaBu/pennyway-was/assets/103185302/2bf50a22-839e-4d50-b3d0-300e3a957966" width = 300/>

### 정리

- 카테고리 삭제 시, 해당 카테고리에 등록한 지출 내역도 함께 삭제한다.
- 미리 사용자에게 삭제될 수 있음을 경고하는 알림을 띄우는데 몇 건의 내역이 존재하는지 함께 표시한다.

**Reply by psychology50** [reply]
희진님..뱃지에 은근 진심이시군요....ㅋㅋㅋㅋㅋㅋ  

1번을 추천하는 이유로 카카오맵을 예시로 들어주셨는데, 서비스의 성격이 다르기 때문에 좀 위험한 접근이라 생각했습니다.  
카카오 맵의 즐겨찾기 그룹은 함께 지운다고 해서, 현재 상태에 별다른 영향을 주지 않아요. 그냥 그런가보다...하겠죠.  

그런데 지출 카테고리를 삭제한다고 지출 내역을 모두 지운다..?  
지출은 분명히 발생한 사건이고, 그렇다면 그 기록들을 관리해야 하는 사용자는 모두 다시 입력을 해야 할까요?  

사실 1번의 방법은 이미 나왔던 이야기지만, 위 의문에 대한 답을 할 수 없었기에 2, 3번의 대안들을 추가로 떠올리게 된 거였습니다.  
희진님의 생각은 어떠한가요?

**Reply by psychology50** [reply]
작업 딜레이를 핑계로 모든 걸 미룰 수는 없습니다.  
항상 말씀드리지만 MVP라는 건, 우리 서비스가 제공하는 기능들이 부실할 수밖에 없는 핑계거리를 위함이 아니라,  
비록 제공하는 기능은 적어도 그 기능만큼은 정책과 동작이 완벽해야 하는 거예요.  

그리고 작업 속도 보면 디자인 팀에서 이 부분 다 만들 때쯤 iOS팀에서 이거 작업 들어가거나, 들어가지 못 하고 있을 확률이 큽니다.  
그래서 그 부분은 배제하고 의견 내주시면 감사드리겠습니다.

+ 추가  
뭔가 혼낸 기분이라 찝찝해서.. `(1)`번을 선택한 이유가 단순히 작업 딜레이가 예상되기 때문만이라면 부적절하다는 의미로 말씀드린 겁니다.

**Reply by BangDori** [reply]
혹시 사용자가 등록한 커스텀 카테고리에 대해서는 수정이 불가능한가요? 피그마에서 찾아봤는데 보이지가 않아서 여쭤봅니닷

**Reply by psychology50** [reply]
예, 저도 그거 말씀드리려고 했는데 이게 먼저 정해져야 플로우를 정리해서 디자인을 요청할 수 있어서 못 하고 있어요

**Reply by BangDori** [reply]
제가 생각하기에는 "커스텀 카테고리의 삭제"를 사용하는 경우는 **2가지 정도**가 있을 것 같아요.

1. 해당 카테고리를 더 이상 사용하지 않을 때
2. 카테고리를 수정하고 싶을 때 (아이콘, 카테고리명)

서비스 성격을 고려한다면 희진님 말씀대로 **카테고리 삭제는 1번을 선택**하고, 추가 기능으로 **카테고리를 수정**할 수 있도록 하면 어떨까요?

**Reply by psychology50** [reply]
서로 다른 카테고리가 2개 존재하며, 각 카테고리에 임의의 개수의 소비 내역이 등록되어 있음을 가정하겠습니다.  
하나는 반드시 사용자 정의 카테고리이며, 다른 하나는 종류가 무엇이든 상관 없습니다.

이제 사용자가 커스텀 카테고리가 불만족스럽다고 느꼈다고 칩시다.  
그럼 사용자는 커스텀 카테고리와 등록된 소비 내역에 3가지 의사 결정을 할 수 있어야 합니다.

1. 커스텀 카테고리에 등록된 모든 소비 내역을 함께 제거한다.
2. 커스텀 카테고리 정보를 수정한다.
3. 커스텀 카테고리는 삭제하되, 기존의 소비 내역은 다른 카테고리로 옮긴다. (카테고리 병합)

어차피 세 기능은 모두 제공되어야 한다고 생각합니다.  
카테고리 삭제는 1번을 택하고, 카테고리를 수정할 때는 2번을 택해서 해결할 수 있습니다.  
하지만 그것만으로는 3번의 유즈 케이스는 여전히 처리할 수 없습니다.

**Reply by psychology50** [reply]
가장 큰 문제는 지출 카테고리 정보를 관리하는 페이지가 없다는 거 같네요. ㅋ ㅋ ㅋ ㅋ ㅋ  하

**Comment by heejinnn** [comment]
넵 저도 단순히 작업 딜레이에 대한 사항만을 문제로 삼은 건 아닙니다! 
작업이 딜레이 되더라도 그만큼 유저에게 좋은 경험이 된다면 해당 경우를 택하는게 옳다고 생각합니다.

```
1. 커스텀 카테고리에 등록된 모든 소비 내역을 함께 제거한다.
2. 커스텀 카테고리 정보를 수정한다.
3. 커스텀 카테고리는 삭제하되, 기존의 소비 내역은 다른 카테고리로 옮긴다. (카테고리 병합)
```

재서님이 정리해주신 내용에서 3번같은 경우, "카테고리 삭제"라는 기능에 "다른 카테고리로 이동" 기능을 추가한 것이라고 생각이 들어요. 

저는 "카테고리 삭제에 `다른 카테고리로 이동`이라는 기능을 추가할 필요가 있을까?", "유저에게 좋은 경험이 되는걸까?"에 대한 의문이 들었어요. 
저의 판단으로는 카테고리를 삭제하는 경우 전체 내용도 삭제된다는 걸 유저가 충분히 인지할 수 있다고 생각을 했고, 
만약 소비내역들을 다른 카테고리로 이동해야 한다면 카테고리를 삭제하는 경우가 아니라 카테고리 수정이나 지출관리 창 내부에서 제공할 수 있는 부분이라고 생각했습니다.

정리하자면, `소비 내역 다른 카테고리로 이동`이라는 기능은 필요하지만, 카테고리 삭제에서 제공할 필요가 있을까에 초점이 맞혀진 것 같습니다.
그래서 유저 경험을 통해 피드백을 받아보고 추후에 3번을 고려해도 된다고 판단했습니다!
